### PR TITLE
[NFC] [TEST] cleanup, more use of return in ActivityTest

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------

[NFC] [TEST] cleanup, more use of return in ActivityTest

Before
----------------------------------------
old

After
----------------------------------------
new

Technical Details
----------------------------------------
I removed some old patterns, added 'return' and removed some redundancy & a test that never actually tested anything

This is one of our earliest test suites & there is a lot in there that is a bit repetitive of this class and others

Comments
----------------------------------------

### CODE REVIEW

File: tests/phpunit/api/v3/ActivityTest.php [hunk 1]
```diff
@@ -1285,36 +1272,12 @@ public function testActivityUpdateWithStatus() {
...
-      'duration' => 120,
-      'location' => '21, Park Avenue',
-      'details' => 'Lets update Meeting',
-      'status_id' => 1,
-      'activity_name' => 'Test activity type',
-      'priority_id' => 1,
-    ];
-
-    $this->callAPISuccess('Activity', 'create', $params);
-    $this->callAPISuccessGetSingle('Activity', ['id' => $activity['id']]);
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/ActivityTest.php:1315]: test never actually does what it says on the box (ie it doesn't check the source) - which SyntaxConformance does

File: tests/phpunit/api/v3/ActivityTest.php [hunk 2]
```diff
@@ -926,56 +925,52 @@ public function testActivityGetGoodIDCustom() {
...
       'contact_id' => $this->_params['source_contact_id'],
       'activity_type_id' => $this->test_activity_type_value,
       'sequential' => 1,
       'return.custom_' . $ids['custom_field_id'] => 1,
     ];
-    $result = $this->callAPIAndDocument('activity', 'get', $params, __FUNCTION__, __FILE__);
-    $this->assertEquals("custom string", $result['values'][0]['custom_' . $ids['custom_field_id']], ' in line ' . __LINE__);
-
-    $this->assertEquals($this->test_activity_type_value, $result['values'][0]['activity_type_id']);
-    $this->assertEquals('test activity type id', $result['values'][0]['subject']);
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ActivityTest.php:947]: I'm not clear why these two lines are removed?

File: tests/phpunit/api/v3/ActivityTest.php [hunk 3]
```diff
@@ -1104,21 +1096,18 @@ public function testActivityUpdateCustom(): void {
...
     $result = $this->callAPISuccess('activity', 'create', $params);
 
     $activityId = $result['id'];
     $result = $this->callAPISuccess($this->_entity, 'get', [
       'return.custom_' . $ids['custom_field_id'] => 1,
-      'version' => 3,
       'id' => $result['id'],
     ]);
-    $this->assertEquals("custom string", $result['values'][$result['id']]['custom_' . $ids['custom_field_id']]);
-    $this->assertEquals("2009-10-18 00:00:00", $result['values'][$result['id']]['activity_date_time']);
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ActivityTest.php:1119]: Also this date one?

File: tests/phpunit/api/v3/ActivityTest.php [hunk 4]
```diff
@@ -1218,28 +1206,27 @@ public function testActivityUpdateCheckCoreFields() {
     $params['assignee_contact_id'] = [$contact3 => $contact3];
     $params['target_contact_id'] = [$contact4 => $contact4];
 
-    $result = $this->callAPISuccess('activity', 'create', $params);
-
-    $this->assertEquals($activityId, $result['id'], ' in line ' . __LINE__);
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ActivityTest.php:1223]: Also this one?

File: tests/phpunit/api/v3/ActivityTest.php [hunk 5]
```diff
@@ -1285,36 +1272,12 @@ public function testActivityUpdateWithStatus() {
       'activity_name' => 'Test activity type',
     ];
 
-    $result = $this->callAPISuccess('activity', 'create', $params);
-    $this->assertEquals($result['id'], $activity['id']);
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ActivityTest.php:1289]: And this one although params are backwards.

File: tests/phpunit/api/v3/ActivityTest.php (hunk 2 above)
Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ActivityTest.php:947]: I pulled out some of the stuff that is repeated in test after test due to apparent copy & paste

File: tests/phpunit/api/v3/ActivityTest.php (hunk 4 above)
Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ActivityTest.php:1223]: Ok - in this case - the call to activity create was removed by accident

Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ActivityTest.php:1223]: Actually no - it was right to remove it - it's not doing an update - it's creating a random other activity that is not checked

File: tests/phpunit/api/v3/ActivityTest.php (hunk 5 above)
Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ActivityTest.php:1289]: in this case just the redundant assert is removed

**Comment by demeritcowboy** [comment]
Ok thanks for the clarification.

**Pull request merged**